lib/replicator.js

```diff
@@ -537,7 +537,7 @@ class Peer {
     const reopen = isRemote === true && this.remoteOpened === true && this.remoteDownloading === false &&
        this.remoteUploading === true && this.replicator.downloading === true
 
-    if (this.useSession && !reopen) this.replicator._closeSession()
+    if (this.useSession && !reopen) this.replicator._closeSessionMaybe()
```

Comment:
Is the intent of this line to ensure the session is closed when `this.remoteOpened === false`? If so, I'd place it inside the if on line 542, to make it obvious. (For the case where the if doesn't trigger, the `this.replicator.removePeer`  already calls `closeSessionMaybe`, although the if there only checks for `this.useSession`, not caring about `reopen`

Reply:
no it should run everytime the channel closes, except when we immediately trigger a reopen